tools.go

```diff
@@ -0,0 +1,33 @@
+// +build tools
+
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This package exists to cause `go mod` and `go get` to believe these tools
+// are dependencies, even though they are not runtime dependencies of any
+// package (these are tools used by our CI builds). This means they will appear
+// in our `go.mod` file, but will not be a part of the build. Also, since the
+// build target is something non-existent, these should not be included in any
+// binaries.
+
+package gax
+
+import (
+	_ "github.com/golang/protobuf/proto"
+	_ "github.com/golang/protobuf/protoc-gen-go"
+	_ "golang.org/x/exp/cmd/apidiff"
+	_ "golang.org/x/lint/golint"
+	_ "golang.org/x/tools/cmd/goimports"
+	_ "honnef.co/go/tools/cmd/staticcheck"
```

Comment:
Looks like this file causes our vendoring tool ([`vndr`](https://github.com/LK4D4/vndr) to produce warnings. Afaics, these imports are not used in the files we vendor, but due to this file warnings are produced (see below).

Wondering if this file could be put in its own (internal?) package?

```
2019/04/14 01:03:20 WARNING: dependency is not vendored: github.com/golang/protobuf/protoc-gen-go
2019/04/14 01:03:20 WARNING: dependency is not vendored: golang.org/x/lint/golint
2019/04/14 01:03:20 WARNING: dependency is not vendored: golang.org/x/tools/cmd/goimports
2019/04/14 01:03:20 WARNING: dependency is not vendored: golang.org/x/tools/imports
2019/04/14 01:03:20 WARNING: dependency is not vendored: golang.org/x/tools/go/ast/astutil
2019/04/14 01:03:21 WARNING: dependency is not vendored: golang.org/x/tools/internal/gopathwalk
2019/04/14 01:03:21 WARNING: dependency is not vendored: golang.org/x/tools/internal/fastwalk
2019/04/14 01:03:21 WARNING: dependency is not vendored: golang.org/x/lint
2019/04/14 01:03:21 WARNING: dependency is not vendored: golang.org/x/tools/go/gcexportdata
2019/04/14 01:03:21 WARNING: dependency is not vendored: golang.org/x/tools/go/internal/gcimporter
2019/04/14 01:03:21 WARNING: dependency is not vendored: github.com/golang/protobuf/protoc-gen-go/generator
2019/04/14 01:03:21 WARNING: dependency is not vendored: github.com/golang/protobuf/protoc-gen-go/grpc
2019/04/14 01:03:21 WARNING: dependency is not vendored: github.com/golang/protobuf/protoc-gen-go/generator/internal/remap
2019/04/14 01:03:21 WARNING: dependency is not vendored: github.com/golang/protobuf/protoc-gen-go/plugin
```